src/outputs/dynamodb_output.rs

```diff
@@ -0,0 +1,308 @@
+    } else if let Ok(bytes) = val.downcast::<PyBytes>() {
+        Ok(AttributeValue::B(Blob::new(bytes.as_bytes())))
+    } else if let Ok(list) = val.downcast::<PyList>() {
+        let attr_list: Result<Vec<AttributeValue>> =
+            list.into_iter().map(create_dynamodb_attribute).collect();
+        Ok(AttributeValue::L(attr_list.unwrap()))
+    } else if let Ok(m) = val.downcast::<PyDict>() {
+        let attribute_map: HashMap<String, AttributeValue> = m
+            .into_iter()
+            .map(|(k, v)| (k.to_string(), create_dynamodb_attribute(v).unwrap()))
+            .collect();
+        Ok(AttributeValue::M(attribute_map))
+    } else {
+        Err(anyhow!("Could not convert `{val:?}` into a DynamoDB type."))
+    }
+}
+
+// impl<'a> FromPyObject<'a> for &'a AttributeValue {
+//     fn extract(obj: &'a PyAny) -> PyResult<Self> {
+//         Ok(<PyBytes as PyTryFrom>::try_from(obj)?.as_bytes())
+//     }
+// }
+
+impl OutputWriter<u64, TdPyAny> for DynamoDBOutput {
+    fn push(&mut self, _epoch: u64, key_payload: TdPyAny) {
+        Python::with_gil(|py| {
+            let (key, payload): (Option<TdPyAny>, TdPyAny) = py_unwrap!(
+                key_payload.extract(py),
+                format!(
+                    "DynamoDBOutput requires a `(key, payload)` 2-tuple; \
+                    got `{key_payload:?}` instead"
+                )
+            );
+            let p: &PyDict = py_unwrap!(
+                payload.extract(py),
+                format!(
+                    "DynamoDBOutput requires that the payload be a Dictionary; got \
+                    `{payload:?}` instead"
+                )
+            );
+
+            // Create the DynamoDB PutItem
+            let mut request = self.client.put_item().table_name(&self.table);
+
+            for (k, v) in p {
+                let key: String = k
+                    .extract()
+                    .expect("Keys must be formatted as Strings for Dynamodb. Got `{k:?}`");
+                let attr = create_dynamodb_attribute(&v).unwrap();
+                request = request.item(key, attr);
+            }
+
+            // Add the key to the request as the primary key for the table
+            if let Some(key) = key {
+                log::debug!("Setting primary key to {key:?}");
+                let attr = AttributeValue::S(key.to_string());
+                request = request.item(&self.primary_key, attr);
+            }
+
+            log::debug!("Finished building request: {request:?}");
+
+            // The DynamoDB crate uses the rust Tracing crate, which
+            // deadlocks here with pyo3-log unless we release the GIL.
+            py.allow_threads(|| {
+                self.rt.block_on(request.send()).unwrap();
+            });
+        });
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use proptest::prelude::*;
+
+    proptest! {
+        #[test]
+        fn test_dynamodb_64_numbers(n: u64) {
```

Comment:
Wondering if it makes sense to have a test calls out the "bool before long" rule